mlir/include/mlir/Dialect/EmitC/IR/EmitCTypes.td

```diff
@@ -76,7 +76,7 @@ def EmitC_ArrayType : EmitC_Type<"Array", "array", [ShapedTypeInterface]> {
 
     static bool isValidElementType(Type type) {
       return type.isIntOrIndexOrFloat() ||
-         emitc::isAnySizeTType(type) ||
+         emitc::isPointerWideType(type) ||
          llvm::isa<PointerType, OpaqueType>(type);
```

Comment:
being pointer-wide doesn't qualify a type for being a vaild array element type. Maybe we want `return emitc::isIntegerIndexOrOpaqueType() || llvm::isa<PointerType>(type);` here?